docs/python-env.md

```diff
@@ -0,0 +1,119 @@
+
+So you could have all your project's normal and dev dependencies installed in one venv, but then have an "overlay" venv in which you install `python-lsp-server` and the plugins you want to use, ideally at versions compatible with your project's main dependencies.
+
+One tool that can do all of this automatically is [uv](https://github.com/astral-sh/uv): Running
+
+```bash
+uv run --with python-lsp-server,pylsp-mypy pylsp
+```
+
+in your project directory creates an ephemeral venv (typically in a location under `~/.cache`) that links to your project's venv but also has `python-lsp-server` and `pylsp-mypy` installed, and then runs `pylsp` inside this virtualenv.
+
+The obvious drawback of this solution is that it is rather complicated and requires special tooling like `uv` or a handwritten script.
+
+
+## Question 2: How to make `pylsp` & plugins run in / use the correct env?
+
+### Option 1 (ok but limited & possibly complicated): Per-plugin configuration of executable / env
+
+As we saw in the initial sections, some plugins come with options to configure the executable to run or environment to work in. As long as you limit yourself to these plugins, you can use these configuration options.
+
+How to use different values for these options in different projects is editor-dependent and may involve some scripting.
+
+### Option 2 (ok): Run editor from inside target venv
+
+If you have an editor that is easy to launch inside a virtualenv (e.g. Vim or Emacs which are typically launched from within a shell) and will use that virtualenv's Python interpreter to run tools like `pylsp`, then you can just do that.
+
+A possible drawback is that this will apply not just to `pylsp` but any other similar Python tooling the editor may use (e.g. Vim plugins written in Python), which may not be desired.
+
+### Option 3 (better): Configure editor to launch `pylsp` in target venv
+
+A better approach may be to configure your editor to launch only `pylsp` in the desired venv.
+
+#### Example: NeoVim
+
+For instance, if you use NeoVim's `vim.lsp` functions to set up the LSP configuration yourself, whether with your own config or a default one provided by [`nvim-lspconfig`](https://github.com/neovim/nvim-lspconfig), you can set/override the command it uses to launch `pylsp` like so:
+
+```vim
+lua << EOF
+  vim.lsp.config('pylsp', {
+    -- Using the approach from Question 1, Option 4 to launch in an "overlay" venv
+    -- on top of the project's venv, if any (NeoVim takes care of sharing LSP servers
+    -- invocations between files in the same project):
+    cmd = { 'uv', 'run', '--with', 'python-lsp-server,pylsp-mypy', 'pylsp' },
+  }
+EOF
+```
+
+See [python-lsp-config#68 (comment)][issue-68-comment] for an example that doesn't use `uv` but tries to determine the project venv manually (e.g. for use with Question 1, Option 3).
+
+[issue-68-comment]: https://github.com/python-lsp/python-lsp-server/pull/68#issuecomment-894650876
```

Comment:
You'll probably ask me to remove this part because it's editor-specific, not sure if there is a better way to demonstrate how to apply some of the advice in this document in a real configuration (or whether that should be done at all).